include/fmt/core.h

```diff
@@ -740,19 +740,32 @@ inline auto get_container(std::back_insert_iterator<Container> it)
 }
 
 template <typename Char, typename InputIt, typename OutputIt>
-FMT_CONSTEXPR auto copy_str(InputIt begin, InputIt end, OutputIt out)
```

Comment:
I think the proper solution would be to constrain this overload to `InputIt` and `OutputIt` types that are different and both not convertible to `const Char*`.

Reply:
It looks to me that the problem you are trying to solve is that this overload is called in cases where it shouldn't. Therefore the easiest fix is to constraint this overload.

Reply:
Want to limit this overload? We have this feature that works in all cases and cannot be limited. And there is an optimization that is sometimes better suited

Reply:
We'll still have an optimization in the other overload (just remove `FMT_ENABLE_IF` there). To clarify, I think the overload set should be something like:

```c++
template <
  typename Char, typename InputIt, typename OutputIt,
  FMT_ENABLE_IF(!is_convertible_v<IntputIt, const Char*> && !is_convertible_v<OutputIt, const Char*>)>
FMT_CONSTEXPR auto copy_str(InputIt begin, InputIt end, OutputIt out);

template <typename Char>
FMT_CONSTEXPR auto copy_str(const Char* begin, const Char* end, Char* out)
    -> Char*;
```

Reply:
Corrected a little. But I cannot understand why we turn off the first overload instead of improving the second.

Reply:
The corrected version doesn't allow `memcpy` version of `copy_str` to be called for wide (exotic) characters. https://godbolt.org/z/z9Kf9jKbY

Reply:
Corrected.